### PR TITLE
Call read barrier if not set to J9_GC_READ_BARRIER_TYPE_NONE

### PR DESCRIPTION
**Pull request by youngar**
* Call region check read barrier in object copy
* Remove redundant read barrier logic in J9MONITORTABLE_OBJECT_LOAD

cc @amicic @dmitripivkine @rwy0717 @charliegracie

### CODE REVIEW

File: runtime/gc_include/ObjectAccessBarrierAPI.hpp [hunk 1]
```diff
@@ -385,6 +385,8 @@ class MM_ObjectAccessBarrierAPI
 				vmThread->javaVM->memoryManagerFunctions->j9gc_objaccess_copyObjectFields(vmThread, objectClass, srcObject, srcOffset, destObject, destOffset);
 			} else if (j9gc_modron_readbar_always == _readBarrierType) {
 				vmThread->javaVM->memoryManagerFunctions->j9gc_objaccess_copyObjectFields(vmThread, objectClass, srcObject, srcOffset, destObject, destOffset);
+			} else if (gc_modron_readbar_region_check == _readBarrierType) {
+				vmThread->javaVM->memoryManagerFunctions->j9gc_objaccess_copyObjectFields(vmThread, objectClass, srcObject, srcOffset, destObject, destOffset);
```
Comment (amicic) [inline runtime/gc_include/ObjectAccessBarrierAPI.hpp]: This inner 'if' branching was (probably) created early with an idea to eventually implement fast inline copy for range_check barrier, but none has been implement and likely won't be in foreseeable future.
I'd suggest to collapse 3 conditional cases in 1 unconditional call to slow path barrier.

Comment (youngar) [reply runtime/gc_include/ObjectAccessBarrierAPI.hpp]: I believe I fixed this.  Does this look like what you had in mind?

Comment (amicic) [reply runtime/gc_include/ObjectAccessBarrierAPI.hpp]: Yes, that's it.

There is similar cleanup you could do for J9MONITORTABLE_OBJECT_LOAD macro.

Comment (youngar) [reply runtime/gc_include/ObjectAccessBarrierAPI.hpp]: I'll fix that too

**Comment by amicic** [comment]
Looks good, now. The title is out of date, though.

**Comment by youngar** [comment]
Updated PR title, but its still not great

**Comment by charliegracie** [comment]
jenkins test sanity plinux jdk8

**Pull request merged**